test/test_typechecker.cpp

```diff
@@ -197,3 +197,76 @@ TEST_SUITE("Error positions for unbound parameters")
+{
+    auto doc = document_fixture{}
+                   .add_global_decl("clock c; double x; void f() { x = true? c : 1.0; }")
+                   .add_default_process()
+                   .parse();
+    CHECK_MESSAGE(doc->get_errors().size() == 0, doc->get_errors()[0].msg);
+    CHECK_MESSAGE(doc->get_warnings().size() == 0, doc->get_warnings()[0].msg);
+}
+
+TEST_CASE("Ternary operator with double and clock")
+{
+    auto doc = document_fixture{}
+                   .add_global_decl("clock c; double x; void f() { x = true? 1.0 : c; }")
+                   .add_default_process()
+                   .parse();
+    CHECK_MESSAGE(doc->get_errors().size() == 0, doc->get_errors()[0].msg);
+    CHECK_MESSAGE(doc->get_warnings().size() == 0, doc->get_warnings()[0].msg);
+}
+
+TEST_CASE("Ternary operator with clock and integer")
+{
+    auto doc = document_fixture{}
+                   .add_global_decl("clock c; double x; void f() { x = true? c : 1; }")
+                   .add_default_process()
+                   .parse();
+    CHECK(doc->get_errors().size() == 1);
+    CHECK_MESSAGE(doc->get_warnings().size() == 0, doc->get_warnings()[0].msg);
+}
+
+TEST_CASE("Ternary operator with clock and bool")
+{
+    auto doc = document_fixture{}
+                   .add_global_decl("clock c; double x; void f() { x = true? c : true; };")
+                   .add_default_process()
+                   .parse();
+    CHECK(doc->get_errors().size() == 1);
+    CHECK_MESSAGE(doc->get_warnings().size() == 0, doc->get_warnings()[0].msg);
+}
+
+TEST_CASE("Ternary operator with clock and clock")
+{
+    auto doc = document_fixture{}
+                   .add_global_decl("clock c; double x; void f() { x = true? c : c; }")
+                   .add_default_process()
+                   .parse();
+    CHECK_MESSAGE(doc->get_errors().size() == 0, doc->get_errors()[0].msg);
+    CHECK_MESSAGE(doc->get_warnings().size() == 0, doc->get_warnings()[0].msg);
+}
+
+TEST_CASE("Ternary operator with constant double")
+{
+    auto doc = document_fixture{}
+                   .add_global_decl("const double VAL = 2;")
+                   .add_global_decl("double x; void f() { x = true? -VAL : VAL; }")
+                   .add_default_process()
+                   .parse();
+    CHECK_MESSAGE(doc->get_errors().size() == 0, doc->get_errors()[0].msg);
+    CHECK_MESSAGE(doc->get_warnings().size() == 0, doc->get_warnings()[0].msg);
+}
+
+TEST_CASE("Ternary operator with constant double and clock")
+{
+    auto doc = document_fixture{}
+                   .add_global_decl("const double VAL = 2;")
+                   .add_global_decl("clock c;")
+                   .add_global_decl("double x; void f() { x = true? -VAL : c; }")
+                   .add_default_process()
+                   .parse();
+    CHECK_MESSAGE(doc->get_errors().size() == 0, doc->get_errors()[0].msg);
+    CHECK_MESSAGE(doc->get_warnings().size() == 0, doc->get_warnings()[0].msg);
+}
```

Comment:
Please add more tests where branched types are `struct`s and arrays (of the same type).
Also add tests where the condition `b` is a logical expression, `int`, `clock` and `double` (expect the conversion to bool by comparing with zero).

It would be cool to support C++ case where the common type can be a reference (not sure if this can be part of this PR or another -- we can postpone):
https://en.cppreference.com/w/cpp/language/operator_other
For example:
```c++
int x;
int y;
void store(bool b, int value) {
   (b ? x : y) = value;
}
```

Reply:
Is there a test for that?
I mean, regression is important (what if somebody breaks it?)

Reply:
I've added a test on the uppaal PR using existing test models